packages/client/src/open-feature.ts

```diff
@@ -80,58 +81,107 @@ export class OpenFeatureAPI extends OpenFeatureCommonAPI {
     await this._provider?.onContextChange?.(oldContext, context);
   }
 
-  setProvider(provider: Provider): OpenFeatureCommonAPI {
+  /**
+   * Sets the provider that OpenFeature will use for flag evaluations of providers without a name.
+   * Setting a provider supersedes the current provider used in new and existing clients without a name.
+   *
+   * @param {Provider} provider The provider responsible for flag evaluations.
+   * @returns {GlobalApi} OpenFeature API
+   */
+  setProvider(provider: Provider): this;
+  /**
+   * Sets the provider that OpenFeature will use for flag evaluations of providers with the given name.
+   * Setting a provider supersedes the current provider used in new and existing clients with that name.
+   *
+   * @param {string} client The name to identify the client
+   * @param {Provider} provider The provider responsible for flag evaluations.
+   * @returns {GlobalApi} OpenFeature API
+   */
+  setProvider(client: string, provider: Provider): this;
```

Comment:
```suggestion
  setProvider(clientName: string, provider: Provider): this;
```

Please update the jsdocs if you agree with this suggestion.

Reply:
Makes sense, I will do.